src/plugins/data/public/query/query_string/language_service/query_result.tsx

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import './_recent_query.scss';
+
+import { BehaviorSubject } from 'rxjs';
+import { EuiButtonEmpty, EuiPopover, EuiText, EuiPopoverTitle } from '@elastic/eui';
+
+import React, { useState, useEffect } from 'react';
+
+export interface QueryStatus {
+  status: string;
```

Comment:
should this be an ENUM? like

`SUCCESS` | `FAILED` | `TIMED_OUT` | `LOADING` | `ABORTED` 

^ examples not the exact list.

Reply:
oh i see can it jsut be this: https://github.com/abbyhu2000/OpenSearch-Dashboards/blob/90d27bbd081e945557e8f66241a4e2d2cfb7c272/src/plugins/discover/public/application/view_components/utils/use_search.ts#L37 and we export it at a higher level

Reply:
added